ui/Screen/IdentityCreation.svelte

```diff
@@ -0,0 +1,175 @@
+<script>
+  import { gql } from "apollo-boost";
+  import { getClient, mutate } from "svelte-apollo";
+  import { pop, push } from "svelte-spa-router";
+  import validatejs from "validate.js";
+
+  import {
+    IMAGE_FILENAME,
+    NAME_MATCH,
+    SINGLE_WORD_MATCH
+  } from "../lib/validationHelpers.js";
+  import { setCurrentIdentity } from "../store/identity.js";
+  import { showNotification } from "../store/notification.js";
+
+  import { Button, Input, Text, Title } from "../DesignSystem/Primitive";
+  import { ModalLayout } from "../DesignSystem/Component";
+
+  let handle,
+    displayName,
+    avatarUrl,
+    validations,
+    beginValidation = false;
+
+  validatejs.options = {
+    fullMessages: false
+  };
+
+  const constraints = {
+    handle: {
+      presence: {
+        message: "You must provide a handle",
+        allowEmpty: false
+      },
+      format: {
+        pattern: SINGLE_WORD_MATCH,
+        message: "Handle should match [a-z0-9][a-z0-9_-]+"
+      }
+    },
+    displayName: {
+      format: {
+        pattern: NAME_MATCH,
+        message: "Display name should match [a-z0-9]"
+      }
+    },
+    avatarUrl: {
+      format: {
+        pattern: IMAGE_FILENAME,
+        message: "Avatar URL should be a valid image filename"
+      }
+    }
+  };
+
+  const validate = () => {
+    if (!beginValidation) {
+      return;
+    }
+    validations = validatejs(
+      {
+        handle: handle,
+        displayName: displayName,
+        avatarUrl: avatarUrl
+      },
+      constraints
+    );
+  };
+
+  $: validate(handle, displayName, avatarUrl);
+
+  const client = getClient();
+
+  const CREATE_IDENTITY = gql`
+    mutation($handle: String!, $displayName: String, $avatarUrl: String) {
+      createIdentity(
+        handle: $handle
+        displayName: $displayName
+        avatarUrl: $avatarUrl
+      ) {
+        id
+        shareableEntityIdentifier
+        metadata {
+          handle
+          displayName
+          avatarUrl
+        }
+      }
+    }
+  `;
+
+  const createIdentity = async () => {
+    let response;
+
+    beginValidation = true;
+    validate();
+
+    if (!validatejs.isEmpty(validations)) return;
+
+    try {
+      response = await mutate(client, {
+        mutation: CREATE_IDENTITY,
+        variables: {
+          handle: handle,
+          displayName: displayName,
+          avatarUrl: avatarUrl
+        }
+      });
+
+      const responseData = response.data.createIdentity;
+      setCurrentIdentity({
+        handle: responseData.metadata.handle,
+        displayName: responseData.metadata.displayName,
+        avatarUrl: responseData.metadata.avatarUrl,
+        shareableEntityIdentifier: responseData.shareableEntityIdentifier
+      });
+      push("/identity/success");
```

Comment:
I'd prefer the whole flow just be one screen, like @MeBrei mentioned previously. But if you still want to have it as separate screens, I'd recommend extracting this path in `path.js`.